### ISSUE TITLE
readGEORawFile() should return a RGChannelSet object

### ISSUE THREAD

**Issue by jordimartorell** [hansenlab/minfi#172]
Hello and thank you for this great package.

I am working with raw 450k data from GEO. readGEORawFile() function is very useful to read the signal intensities files provided by GEO, which are the way used by GEO to provide raw data. However, I wonder why readGEORawFile() returns a GenomicMethylSet object instead of a RGChannelSet object.

If I understood well the documentation, GenomicMethylSet is a class designed to store processed data, but we are reading raw GEO data. This is a problem because I can't use functions like preprocessNoob() or detectionP() in order to process these raw data. In my opinion, it would be much more convenient if readGEORawFile() could return a RGChannelSet object.

Thanks in advance!
Jordi

**Comment by kasperdanielhansen** [comment]
But the raw signal intensities provided by GEO are not the same as the
(even lower level) data contained in the IDAT files.  If the IDAT files are
available you should download those and parse them.  It might be possible
to automate this process.

Best,
Kasper

**Comment by jordimartorell** [comment]
Thanks for your response @kasperdanielhansen. Unfortunately, idat files are not usually provided, so we have to start our analysis from the signal intensities files, which I think are the mandatory raw files to upload 450k data to GEO.

**Comment by kasperdanielhansen** [comment]
Yeah, so the issue is that these signal intensities are not all the data.
They don't include
  - control probes
  - "out-of-band" probes
Both of which are required for Noob and the first of which are required for
detectionP.

Best,
Kasper

**Comment by jordimartorell** [comment]
Then I suppose nothing can be done. I don't understand why GEO doesn't require to upload IDAT files, which are the necessary data to reproduce any analysis.
Anyway, thank you very much for your explanations, Kasper.

Best.
Jordi

**Comment by somethingp** [comment]
A lot of the geo raw data does have control probes and detectionP's, however. Look at GSE31848 as an example. I think minfi needs to have some way to deal with this data, or at least provide a way to load this data in.

**Comment by oleksii-nikolaienko** [comment]
Hi,
I'd also like to thank for a great package and ask similar question: I'm starting analysis with text file containing raw values, because IDAT files are not available (GSE40279). After reading with readGEORawFile it is not possible to do preprocessQuantile, which returns:
```
geo.raw <- readGEORawFile(filename="GSE40279_signal_intensities.txt", sep="\t", Uname="SignalA", Mname="SignalB", row.names=2)
genomic.ratio.set   <- preprocessQuantile(geo.raw, mergeManifest=TRUE, fixOutliers=FALSE)
[preprocessQuantile] Mapping to genome.
Error in do_one(nmeth) : NA/NaN/Inf in foreign function call (arg 1)
In addition: Warning message:
In preprocessQuantile(idat.data, mergeManifest = TRUE, fixOutliers = FALSE) :
  preprocessQuantile has only been tested with 'preprocessRaw'
```
Is it because preprocessQuantile requires IDAT files or because something is wrong with that text file?
Of note, there are no NA values in either Meth or Unmeth, but maybe preprocessQuantile somewhere computes beta values without offset, which results in NaNs?

Best,
Oleksii
